### PR TITLE
feat: Implement Satscomma Formatting for Sats Amounts

### PR DESCRIPTION
**Pull request by thelmuxkriovar**
**Description:**

This PR introduces the satcomma standard for formatting sats amounts. You can read [Bitcoin Magazine's 2021 article](https://bitcoinmagazine.com/culture/satcomma-standard-look-at-bitcoin-this) on the subject if you're not familiar with it.

The satcomma standard promotes readability and clarity when dealing with satoshis, by inserting a separator such as spaces or commas every three digits, starting from the decimal point for BTC values or from the least significant digit for satoshi amounts.

**Reasoning:**

- **User Experience**: Improving the way sats amounts are displayed can enhance the user experience, making it easier for users to verify amounts at a glance without counting digits
- **Standardization** - Satscomma is a standard recommended by [bitcoin.design](https://bitcoin.design/guide/designing-products/units-and-symbols/#satcomma)
- **Future-Proof**: As Bitcoin's adoption grows, and transactions in satoshis become more common, having a clear and consistent formatting standard will become increasingly important.

### CODE REVIEW

File: src/utils/denomination.ts [hunk 1]
```diff
@@ -51,10 +51,21 @@ export const formatAmountDenomination = (
             return amountBig.toString();
 
         default:
-            return amount.toString();
+            return satsComma(amount.toString());
     }
 };
 
+function satsComma(sats: string): string {
```
Comment (michael1011) [inline src/utils/denomination.ts:58]: Could you an arrow function here?

**Review by michael1011 — APPROVED** [review]
LGTM! Thank you for your contribution!

One minor change and running `npm run prettier` and this should be good to merge

**Review by kilrau — CHANGES_REQUESTED** [review]
Thank you so much for you contribution! This actually closes an issue which I attached to this PR: https://github.com/BoltzExchange/boltz-web-app/issues/146 :100: 

Tested, looks good! Tests need a fix, then good to go!

File: tests/utils/denomination.spec.ts [hunk 2]
```diff
@@ -78,7 +78,7 @@ describe("denomination utils", () => {
         test.each`
             denomination         | amount                  | valid
             ${denominations.sat} | ${"123123"}             | ${true}
-            ${denominations.sat} | ${"12312300000"}        | ${false}
```
[the remark this answers is not in the document]
Comment (michael1011) [reply tests/utils/denomination.spec.ts:81]: You increased the number of digits: https://github.com/BoltzExchange/boltz-web-app/pull/494/files#diff-dc22175eeda6fbd73202ad7b56a3ad1122f4e277d858d8950e2ddf90655ff6c4R90

[the remark this answers is not in the document]
Comment (kilrau) [reply tests/utils/denomination.spec.ts:81]: Revert this change and you should be good

**Comment by dni** [comment]
just as a reference is tried something similar here, https://github.com/BoltzExchange/boltz-web-app/pull/165

**Comment by kilrau** [comment]
Can you run prettier again? CI still complaining... @thelmuxkriovar 

image

PS: Replacing one or multiple (mark) numbers in the middle of a satcomma formatted amount makes the cursor jump to the very right. This is an acceptable trade-off imo, https://github.com/BoltzExchange/boltz-web-app/pull/165 tried to fix these edge cases and ended up being overly complicated and buggy.

File: tests/utils/denomination.spec.ts [hunk 3]
```diff
@@ -52,18 +52,18 @@ describe("denomination utils", () => {
     describe("calculate allowed digits", () => {
         test.each`
             denomination         | digits | amount
-            ${denominations.sat} | ${5}   | ${1000}
-            ${denominations.sat} | ${7}   | ${100000}
-            ${denominations.sat} | ${10}  | ${100000000}
-            ${denominations.sat} | ${11}  | ${1000000000}
-            ${denominations.sat} | ${12}  | ${10000000000}
+            ${denominations.sat} | ${6}   | ${1000}
```
Comment (dni) [inline tests/utils/denomination.spec.ts:55]: this should be 5 digits `1 000` some goes for following tests

**Review by kilrau — APPROVED** [review]

**Review by dni — APPROVED** [review]
LGTM! thanks!

**Review by michael1011 — APPROVED** [review]
We took the liberty of wrapping up the PR

Thank you!

**Comment by kilrau** [comment]
> Awesome, sorry I haven't been very responsive I've been taking this week a bit slowly coding wise :)

Not at all, this was pretty smooth and we are planning to tag a release and deploy this soon!

Btw what motivated you to throw up this PR? Are you a Boltz Web App user and were missing this feature?

**Comment by kilrau** [comment]
Very nice! Any other issue or feature you would want to work on in exchange for some sats - hmu (my handle is the same on all major platforms)!
